frontend/views/pages/GroupDashboard.vue

```diff
@@ -107,10 +114,25 @@ export default ({
     humanDate,
     handleIncomeClick (e) {
       sbp('okTurtles.events/emit', OPEN_MODAL, 'IncomeDetails')
+
+      // firstly, make sure the listener is attached to 'GROUP_PROFILE_UPDATE' more than twice.
+      // e.g. user dismiss the modal without actually 'updating the income-details' and then click on the banner again.
+      sbp('okTurtles.events/off', GROUP_PROFILE_UPDATE, this.closeBanner)
+      sbp('okTurtles.events/once', GROUP_PROFILE_UPDATE, this.closeBanner)
     },
     closeBanner () {
-      localStorage.setItem('giHideDistributionWarning', true)
+      localStorage.setItem(this.bannerStorageKey, true)
       this.ephemeral.hideBanner = true
+    },
+    checkIfBannerBeenDismissed () {
```

Comment:
`checkIfBannerBeenDismissed` isn't quite correct English, and also it is doing more than simply checking, it is also updating the value - potentially dismissing the banner.

Another possible name that might be a bit clearer would be: `updateBannerVisibility`

Reply:
@taoeffect yeah that sounds better. coming up with the best name is the hardest part in writing a JS code btw lol.